python/kserve/test/test_server.py

```diff
@@ -73,7 +77,17 @@ def load(self):
         self.ready = True
 
     async def predict(self, request, headers=None):
-        return {"predictions": request["instances"]}
+        if isinstance(request, InferRequest):
+            infer_input = request.to_rest()
+            response_id = generate_uuid()
+            result = infer_input["inputs"][0]["data"]
+            infer_output = InferOutput(name="output-0", shape=list(
+                infer_input["inputs"][0]["shape"]), datatype=infer_input["inputs"][0]["datatype"], data=result)
+            infer_response = InferResponse(model_name=self.name, infer_outputs=[
+                                           infer_output], response_id=response_id)
+            return infer_response.to_rest()
+        else:
+            return {"predictions": request["instances"]}
```

Comment:
this code is repeated in the below section (starting with line 107)- can we move this code to a helper method to avoid repeating it?